src/main/kotlin/notification/application/NotificationCreator.kt

```diff
@@ -0,0 +1,36 @@
+package notification.application
+
+import deadline.domain.Time
+import note.application.NoteSearcher
+import note.domain.NoteRepository
+import note.domain.exceptions.NonExistentNoteException
+import notification.domain.Notification
+import notification.domain.NotificationRepository
+import notification.domain.exceptions.AlreadyConfiguredNotification
+import shared.domain.Identifier
+import shared.domain.exceptions.InvalidUUIDException
+
+class NotificationCreator(
+    private val repository: NotificationRepository,
+    private val noteSearcher: NoteRepository
```

Comment:
Should be named noteRepository, since noteSearcher is the name of the use case:
```suggestion
    private val noteRepository: NoteRepository
```